pyccel/codegen/printing/ccode.py

```diff
@@ -105,6 +107,18 @@ def _traverse_matrix_indices(self, mat):
 
     # ============ Elements ============ #
 
+    def _print_PythonFloat(self, expr):
+        value = self._print(expr.arg)
+        return '(double)({0})'.format(value)
+
+    def _print_Int(self, expr):
+        value = self._print(expr.arg)
+        return '(long)({0})'.format(value)
+
+    def _print_Bool(self, expr):
+        value = self._print(expr.arg)
+        return '{} ? 1 : 0'.format(value)
```

Comment:
We should differentiate between different types. E.g.:
**x is bool**
```
x = (a == 2)
if (bool(x)):
```
will give:
```
if (x ? 1 : 0)
```
when
```
if (x)
```
would be sufficient

**x is int**
```
if (bool(x)):
```
will give:
```
if (x ? 1 : 0)
```
when
```
if (x)
```
would be sufficient. Although the former has the advantage of ensuring that a boolean is always one of `0` and `1` so it may be clearer to give:
```
if (x != 0)
```
Do you have opinions on the better formulation @saidctb @yguclu ?

Reply:
C99 provides the <stdbool.h> header with the `bool` data type, and `true` and `false` constants. Maybe using them would add a bit of clarity to the printed code?

In the former case I would write
```C
bool x = (a == 2);

if (x)
{
  ...
}
```
and in the latter case I would not use implicit casting to bool in order to avoid unpleasant surprises:
```C
if (x != 0)
{
  ...
}
```

Reply:
the suggested solution of @yguclu seems good to me